ledger-core/virtual/execute/execute_test.go

```diff
@@ -562,4 +562,53 @@ func TestSendVStateReportWithMissingState_IfConstructorWasInterruptedBeforeRunne
 		close(vStateReportRecv)
 		t.Error("timeout")
 	}
+
+	mc.Finish()
+}
+
+func TestSMExecute_StopWithoutMessagesIfPulseChangedBeforeOutgoing(t *testing.T) {
+	t.Log("C5101")
+	var (
+		mc  = minimock.NewController(t)
+		ctx = context.Background()
+
+		class                              = gen.UniqueGlobalRef()
+		caller                             = gen.UniqueGlobalRef()
+		catalog     object.Catalog         = object.NewLocalCatalog()
+		authService authentication.Service = authentication.NewServiceMock(t)
+	)
+
+	slotMachine := slotdebugger.New(ctx, t)
+	slotMachine.PrepareRunner(ctx, mc)
+
+	slotMachine.AddInterfaceDependency(&catalog)
+	slotMachine.AddInterfaceDependency(&authService)
+
+	slotMachine.InitEmptyMessageSender(mc)
+
+	outgoing := slotMachine.GenerateLocal()
+
+	smExecute := SMExecute{
+		Payload: &payload.VCallRequest{
+			CallType:     payload.CTConstructor,
```

Comment:
Test case about terminal method, why are you using CTConstructor?

Reply:
Fix call type